### PR TITLE
TECH Add tests

### PR DESCRIPTION
**Pull request by gcoupelant**
Some litttle refactoring (better exception handling, and more) of `docker`, `git` and `github`, and add tests to those modules

### CODE REVIEW

File: src/lib/docker.py [hunk 1]
```diff
@@ -116,8 +120,8 @@ def generate_app_compose_templates(env, app_name, app, docker_image=None):
         return
 
     if not os.path.exists(os.path.join(working_dir_current_app, 'Dockerfile')):
```
Comment (adrien-dong) [inline src/lib/docker.py]: Maybe test if docker image exists instead of Dockefile

File: src/lib/docker.py [hunk 2]
```diff
@@ -79,13 +83,13 @@ def generate_app_common_file(app, app_dir, env, docker_image):
             ]
```
Comment (adrien-dong) [inline src/lib/docker.py]: Not in this PR, but apps should always have a docker image in Docker hub

File: src/lib/docker.py [hunk 3]
```diff
@@ -141,14 +145,16 @@ def get_app_processes_lists(env, app_name, app):
     if procfile_data is None:
         raise ValueError('App %s has no procfile data' % app_name)
 
+    if not os.path.exists(os.path.join(working_dir, 'Dockerfile')):
```
Comment (adrien-dong) [inline src/lib/docker.py]: same as above test docker image instead of Dockerfile

File: src/lib/docker.py (hunk 1 above)
Comment (gcoupelant) [inline src/lib/docker.py]: see https://github.com/transcovo/letsdeploy/issues/35

File: src/lib/docker.py (hunk 3 above)
Comment (gcoupelant) [inline src/lib/docker.py]: see https://github.com/transcovo/letsdeploy/issues/35

**Pull request merged**